[Bug]: Nightly / CI failed - tests/diffusion/distributed/test_wan_spatial_shard.py::test_spatial_shard_decode_matches_reference[height] - AssertionError: spatial_shard_height max_abs_diff 0.8535196781158447 exceeds 0.03

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
ci env
```

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
0.27.0
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
6f55a06fab90
```
</details>

### 🐛 Describe the bug

**Buildkite build:** https://buildkite.com/vllm/vllm-omni/builds/13832
**Buildkite step:** [:full_moon: Diffusion · Distributed Test with H100](https://buildkite.com/vllm/vllm-omni/builds/13832#01a01c7b-76c4-4aa9-954e-46dda2034083)

**Failure kind:** pytest FAILED
**Test node:** `tests/diffusion/distributed/test_wan_spatial_shard.py::test_spatial_shard_decode_matches_reference[height]`

**Log reason:**
AssertionError: spatial_shard_height max_abs_diff 0.8535196781158447 exceeds 0.03

**Analysis:**
Category: assertion failure. Expected vs actual mismatch — check test logic, edge cases, and data setup.

**Error log excerpt:**

```text
split_dim = 'height'



    @pytest.mark.full_model

    @pytest.mark.diffusion

    @pytest.mark.parallel

    @hardware_test(res={"cuda": "H100"}, num_cards=_SPATIAL_SHARD_WORLD_SIZE)

    @pytest.mark.parametrize("split_dim", ["height", "width"])

    def test_spatial_shard_decode_matches_reference(split_dim: str):

        manager = mp.get_context("spawn").Manager()

        return_dict = manager.dict()

        # Use a per-split-dim port to avoid collisions across parametrized runs.

        master_port = str(29500 + (1 if split_dim == "width" else 0))

    

        mp.spawn(

            _spatial_shard_decode_worker,

            args=(split_dim, return_dict, master_port),

            nprocs=_SPATIAL_SHARD_WORLD_SIZE,

            join=True,

        )

    

        assert "max_abs_diff" in return_dict, "rank 0 did not report a result"

        max_abs_diff = return_dict["max_abs_diff"]

        mean_abs_diff = return_dict["mean_abs_diff"]

        print(

            f"spatial_shard_{split_dim} vs reference: max_abs_diff={max_abs_diff:.6e} "

            f"mean_abs_diff={mean_abs_diff:.6e} shape={return_dict.get('shape')}"

        )

>       assert max_abs_diff <= _SPATIAL_SHARD_TOLERANCE, (

            f"spatial_shard_{split_dim} max_abs_diff {max_abs_diff} exceeds {_SPATIAL_SHARD_TOLERANCE}"

        )

E       AssertionError: spatial_shard_height max_abs_diff 0.8535196781158447 exceeds 0.03

E       assert 0.8535196781158447 <= 0.03



tests/diffusion/distributed/test_wan_spatial_shard.py:563: AssertionError
```

---
*Generated from a nightly HTML report. Redact secrets before submitting; complete the checkboxes on GitHub.*

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.

--- a/tests/diffusion/distributed/test_distributed_vae_executor.py
+++ b/tests/diffusion/distributed/test_distributed_vae_executor.py
@@ -1,3 +1,4 @@
+from dataclasses import dataclass
 from types import SimpleNamespace
 
 import pytest
@@ -58,6 +59,11 @@ class E2EOperator:
         return torch.cat(tiles, dim=0)
 
 
+@dataclass
+class FakeWorldGroup:
+    device_group: object
+
+
 class DummyMixin(DistributedVaeMixin):
     def __init__(self):
         self.use_tiling = True
@@ -75,6 +81,16 @@ def mock_dist(monkeypatch: pytest.MonkeyPatch):
 
 
 @pytest.fixture(autouse=True)
+def mock_world_group(monkeypatch: pytest.MonkeyPatch):
+    group = object()
+    monkeypatch.setattr(
+        "vllm_omni.diffusion.distributed.autoencoders.distributed_vae_executor.get_world_group",
+        lambda: FakeWorldGroup(device_group=group),
+    )
+    return group
+
+
+@pytest.fixture(autouse=True)
 def mock_dist_vae_executor(monkeypatch: pytest.MonkeyPatch):
     monkeypatch.setattr(DistributedVaeExecutor, "gather_tensors", lambda self, x: [x])
     monkeypatch.setattr(DistributedVaeExecutor, "broadcast_tensor", lambda self, x: x)
@@ -83,6 +99,11 @@ def mock_dist_vae_executor(monkeypatch: pytest.MonkeyPatch):
 # ============================
 # Unitest
 # ============================
+
+
+def test_uses_dedicated_world_device_group(mock_world_group):
+    executor = DistributedVaeExecutor()
+    assert executor.group is mock_world_group
 
 
 def test_balance_tasks():

--- a/vllm_omni/diffusion/distributed/autoencoders/distributed_vae_executor.py
+++ b/vllm_omni/diffusion/distributed/autoencoders/distributed_vae_executor.py
@@ -9,6 +9,8 @@ import torch
 import torch.distributed as dist
 from vllm.logger import init_logger
 
+from vllm_omni.diffusion.distributed.parallel_state import get_world_group
+
 logger = init_logger(__name__)
 
 
@@ -43,10 +45,10 @@ class DistributedVaeExecutor:
     """
 
     def __init__(self):
-        # VAE tiles are scheduled across the complete worker WORLD.
-        self.group = None
-        self.world_size = dist.get_world_size()
-        self.rank = dist.get_rank()
+        # Use a dedicated process group spanning the complete worker WORLD.
+        self.group = get_world_group().device_group
+        self.world_size = dist.get_world_size(self.group)
+        self.rank = dist.get_rank(self.group)
         self.parallel_size = 1
         self.parallel_mode = "tile"
 

